utils/dateUtils.ts

```diff
@@ -0,0 +1,36 @@
+export const formatWithTimeAndDate = (isoString: Date) => {
+    if (isoString) {
+        // Parse the ISO date string into a Date object
+        const date = new Date(isoString);
+
+        // Define arrays for month names
+        const months = ["Jan", "Feb", "Mar", "Apr", "May", "Jun", "Jul", "Aug", "Sep", "Oct", "Nov", "Dec"];
+
+        // Extract date components
+        const hours = date.getUTCHours();
+        const minutes = date.getUTCMinutes();
+        const day = date.getUTCDate();
+        const month = date.getUTCMonth(); // Month is 0-based
+        const year = date.getUTCFullYear();
```

Comment:
Should we use locale time in the UI instead of UTC? I suspect that most users would prefer it

Reply:
Oops. Thanks for catching that @briri .

Reply:
it prompted me fix the dates in the backend 🙃